products/dialogflow/api.yaml

```diff
@@ -0,0 +1,123 @@
+# Copyright 2020 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Dialogflow
+display_name: Dialogflow
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://dialogflow.googleapis.com/v2/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+apis_required:
+  - !ruby/object:Api::Product::ApiReference
+    name: Dialogflow API
+    url: https://console.cloud.google.com/apis/library/dialogflow.googleapis.com
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Agent'
+    base_url: "projects/{{project}}/agent"
+    self_link: "projects/{{project}}/agent"
+    update_verb: :POST
+    description: |
+      Represents a Dialogflow agent.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://cloud.google.com/dialogflow/docs/'
+      api: 'https://cloud.google.com/dialogflow/docs/reference/rest/v2/projects/agent'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'displayName'
+        required: true
+        input: true
```

Comment:
Changes to DisplayName forces recreation?

Reply:
It does not, good catch